service/csv/csv_worker.go

```diff
@@ -0,0 +1,125 @@
+package csv
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/rodrigomkd/go-rest-api/model"
+)
+
+type ICSVWService interface {
+	ReadWorkers() []model.Worker
+}
+
+type CSVWorkerService struct {
+	csv            ICSVWService
+	dataSourcePath string
+}
+
+func NewWorker(dataSourcePath string) *CSVWorkerService {
+	return &CSVWorkerService{
+		dataSourcePath: dataSourcePath,
+	}
+}
+
+func (cs CSVWorkerService) ReadWorkers(typ string, items int, itemsPerWork int) []model.Worker {
+	f, _ := os.Open(cs.dataSourcePath)
+	defer f.Close()
+
+	ts := time.Now()
+	activities := cs.concuRSwWP(f, typ, items, itemsPerWork)
+	te := time.Now().Sub(ts)
+
+	fmt.Println("END Concurrency: ", te)
+
+	return activities
+}
+
+// with Worker pools
+func (cs CSVWorkerService) concuRSwWP(f *os.File, typ string, items int, itemsPerWork int) []model.Worker {
+	fcsv := csv.NewReader(f)
+	rs := make([]model.Worker, 0)
```

Comment:
You could initialize this to the number of items you should return since we already know that value. That way you wouldn't have to make an `append`